app/src/main/java/com/android/sample/ui/recipeOverview/RecipeOverview.kt

```diff
@@ -45,323 +39,454 @@ import androidx.compose.ui.graphics.Color
+                  Modifier.weight(1f),
+                  clickable = { isIngredientDisplay = true })
+
+              SlidingButton(
+                  isIngredientDisplay,
+                  stringResource(R.string.instructions),
+                  "instructionsButton",
+                  Modifier.weight(1f),
+                  clickable = { isIngredientDisplay = false })
             }
+
+        // Display the appropriate view based on the toggle state
+        IngredientInstructionView(isIngredientDisplay, currentRecipe)
       }
 }
 
-/** Display of the recipe image */
+/**
+ * Custom button that slides to the left or right based on the selection
+ *
+ * @param isIngredientDisplay: Boolean to determine if the ingredient is displayed
+ * @param text: Text to display on the button
+ * @param testTag: Test tag for the button
+ * @param modifier: Modifier to apply to the button
+ */
 @Composable
-private fun RecipeImage(currentRecipe: Recipe?, width: Dp, height: Dp) {
+private fun SlidingButton(
+    isIngredientDisplay: Boolean,
+    text: String,
+    testTag: String,
+    modifier: Modifier = Modifier,
+    clickable: () -> Unit
+) {
+  Box(
+      modifier =
+          modifier
+              .background(
+                  if (!isIngredientDisplay) MaterialTheme.colorScheme.onBackground
+                  else Color.Transparent,
+                  shape = RoundedCornerShape(IMAGE_ROUND_CORNER.dp))
+              .clickable { clickable() }
+              .padding(vertical = SMALL_PADDING.dp),
+      contentAlignment = Alignment.Center) {
+        Text(
+            text = text,
+            style = MaterialTheme.typography.bodyMedium,
+            color =
+                if (!isIngredientDisplay) MaterialTheme.colorScheme.background
+                else MaterialTheme.colorScheme.onPrimary,
+            textAlign = TextAlign.Center,
+            modifier = Modifier.padding(SMALL_PADDING.dp).testTag(testTag))
+      }
+}
+
+/**
+ * Display of the recipe image, title, rating and category
+ *
+ * @param currentRecipe: The recipe to display
+ */
+@Composable
+private fun RecipeImage(currentRecipe: Recipe) {
+  val height = LocalConfiguration.current.screenHeightDp.dp * OVERVIEW_RECIPE_RATE
+
   Card(
-      modifier = Modifier.fillMaxWidth().padding(8.dp),
-      shape = RoundedCornerShape(16.dp),
-      elevation = CardDefaults.cardElevation(4.dp)) {
+      modifier = Modifier.fillMaxWidth().padding(SMALL_PADDING.dp),
+      shape = RoundedCornerShape(OVERVIEW_RECIPE_CARD_SHAPE.dp),
+      elevation = CardDefaults.cardElevation(OVERVIEW_RECIPE_CARD_ELEVATION.dp)) {
         Column(modifier = Modifier.background(color = MaterialTheme.colorScheme.onPrimary)) {
           Image(
-              painter = rememberAsyncImagePainter(model = currentRecipe?.strMealThumbUrl),
+              painter = rememberAsyncImagePainter(model = currentRecipe.strMealThumbUrl),
               contentDescription = "Recipe Image",
```

Comment:
**Content Description**:
   - Instead of hardcoding `"Recipe Image"` directly in `contentDescription`, it’s a good practice to add it to `strings.xml`. This way, it’s easier to manage and supports localization if needed.

   **Suggestion**:
   ```kotlin
   contentDescription = stringResource(R.string.recipe_image_description)
   ```

   And then add it to `strings.xml`:
   ```xml
   <string name="recipe_image_description">Recipe Image</string>
   ```

Reply:
Good catch ! I've missed this one.